### DISCUSSION TITLE
Field \"_in\" is not defined by type XXX

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#7301] [Archive]
Hi everyone.
I am seeing the following error. Could anyone tell me what to do?

Thanks!
```
{
   "errors":[
      {
         "message":"GraphQL validation error.",
         "extensions":{
            "code":"GRAPHQL_VALIDATION_EXCEPTION",
            "graphqlErrors":[
               {
                  "message":"Field \"_in\" is not defined by type \"action_item_filter\". Did you mean \"_and\", \"_or\", or \"id\"?",
                  "locations":[
                     {
                        "line":3,
                        "column":54
                     }
                  ]
               }
            ]
         }
      }
   ]
}
```

**Comment by rijkvanzanten** [comment]
From the error, it sounds like `action_item` is a relational field, meaning you can only use _and, _or, or the field to filter on on this level (id)

**Reply by powerseed** [reply]
Would you please provide an example about how to use `the field` to filter?
Thank you so much!

**Reply by rijkvanzanten** [reply]
What were you trying to do? Filtering where `action_item` is `_in` `1, 2, 3`? 

If that's the case, then 

```
filter: {
  article_item: {
    id: {
      _in: [1, 2, 3]
    }
  }
}
```

**Reply by powerseed** [reply]
Sorry I still don't quite understand.

I have a table `action_item_content`, in which there is a field `action_item_id`, which is a FK to another table `action_item`.

So in my query for `action_item_content` objects, I would like to filter by this `action_item_id` field. 

My query is 
```graphql
query($actionItemId: [Float]) {
  action_item_content(filter: { action_item: { id: { _in: $actionItemId } } }) {
    id
    action_item_id {
      id
    }
  }
}
```
But I am seeing error 
`Field \"action_item\" is not defined by type \"action_item_content_filter\". Did you mean \"action_item_id\"?`

Please help me! Thanks in advance!

**Reply by rijkvanzanten** [reply]
> in which there is a field action_item_id, which is a FK to another table action_item.

Gotcha. GraphQL doesn't do union types, so you can only filter on the nested fields under `action_item_id`.

Try using

```graphql
query($actionItemId: [Float]) {
  action_item_content(filter: { action_item_id: { id: { _in: $actionItemId } } }) {
    id
    action_item_id {
      id
    }
  }
}
```

**Reply by powerseed** [reply]
It works! 
I didn't understand how filter works. 
Thank you for helping me!
